firmware/Core/Src/unit_tests/test_telecommand_parser.c

```diff
@@ -122,3 +122,25 @@ uint8_t TEST_EXEC__TCMD_ascii_to_double() {
 
     return 0;
 }
+
+uint8_t TEST_EXEC_TCMD_parse_full_telecommand() {
+    TCMD_parsed_tcmd_to_execute_t parsed_tcmd;
+    // Testing correct usage of 0 argument telecommand 
+    TEST_ASSERT_TRUE(TCMD_parse_full_telecommand("CTS1+hello_world()!", TCMD_TelecommandChannel_DEBUG_UART, &parsed_tcmd) == 0);
+    TEST_ASSERT_TRUE(strcmp(parsed_tcmd.args_str_no_parens, "") == 0);
+    TEST_ASSERT_TRUE(parsed_tcmd.tcmd_channel == TCMD_TelecommandChannel_DEBUG_UART);
+    TEST_ASSERT_TRUE(strcmp(TCMD_telecommand_definitions[parsed_tcmd.tcmd_idx].tcmd_name, "hello_world") == 0);
+
+    // Tesing incorrect usage of 0 argument telecommand
+    TEST_ASSERT_TRUE(TCMD_parse_full_telecommand("CTS1+hello_world(a)!", TCMD_TelecommandChannel_DEBUG_UART, &parsed_tcmd) == 100);
+    TEST_ASSERT_TRUE(TCMD_parse_full_telecommand("CTS1+hello_world(a,1)!", TCMD_TelecommandChannel_DEBUG_UART, &parsed_tcmd) == 100);
+
+    // Testing correct usage of 1 argument telecommand 
+    TEST_ASSERT_TRUE(TCMD_parse_full_telecommand("CTS1+set_system_time(10)!", TCMD_TelecommandChannel_DEBUG_UART, &parsed_tcmd) == 0); 
+    
+    // Tesing incorrect usage of 1 argument telecommand
+    TEST_ASSERT_TRUE(TCMD_parse_full_telecommand("CTS1+set_system_time()!", TCMD_TelecommandChannel_DEBUG_UART, &parsed_tcmd) == 100);
+    TEST_ASSERT_TRUE(TCMD_parse_full_telecommand("CTS1+set_system_time(10,1)!", TCMD_TelecommandChannel_DEBUG_UART, &parsed_tcmd) == 100);
+    
```

Comment:
Please add the following test cases for which the validation should fail. I _think_ they're all caught right now, but still a good check.

```
CTS1+hello_world(,)!

// Edge cases: commas at starts and ends
CTS1+set_system_time(123,)!
CTS1+set_system_time(,123)!

// flash_read_hex is a 3-arg function
CTS1+flash_read_hex(0,0,0)! // Expected success
CTS1+flash_read_hex(0,0,0,)! // Expected fail
CTS1+flash_read_hex(,0,0,0)! // Expected fail
CTS1+flash_read_hex(0,0,0,)! // Expected fail
CTS1+flash_read_hex(0,0)! // Expected fail
```